dan_layer/wallet/storage_sqlite/src/reader.rs

```diff
@@ -315,6 +315,21 @@ impl WalletStoreReader for ReadTransaction<'_> {
             .collect::<Result<_, _>>()?;
         Ok(outputs)
     }
+
+    fn get_proof_by_transaction_hash(
+        &mut self,
+        transaction_hash: &str,
```

Comment:
nit:
```suggestion
        transaction_hash: FixedHash,
```

The reason we do this is (1) to make sure the data is always well formed and (2) another database implementation may want to use 32 bytes rather than a hex representation